great_expectations/util.py

```diff
@@ -1298,19 +1304,19 @@ def is_truthy(value: Any) -> bool:
         return False
 
 
-def is_numeric(value: Any) -> bool:
+def is_numeric(value: Any) -> TypeGuard[Union[int, float]]:
     return value is not None and (is_int(value=value) or is_float(value=value))
 
 
-def is_int(value: Any) -> bool:
+def is_int(value: Any) -> TypeGuard[int]:
     try:
         int(value)
     except (TypeError, ValueError):
         return False
     return True
 
 
-def is_float(value: Any) -> bool:
+def is_float(value: Any) -> TypeGuard[float]:
```

Comment:
On closer inspection this doesn't work as a `TypeGuard`, it's checking that that value is convertible to an `int` or `float` but not that it actually is.
For example `"1.5"` would pass the `is_float` test, but it's still a `str`.